Source/WebCore/Modules/encryptedmedia/MediaKeySession.cpp

```diff
@@ -495,6 +502,7 @@ void MediaKeySession::close(Ref<DeferredPromise>&& promise)
             m_taskQueue.enqueueTask([this, promise = WTFMove(promise)] () mutable {
                 // 5.3.1. Run the Session Closed algorithm on the session.
                 sessionClosed();
+                m_closed = true;
```

Comment:
sessionClosed sets this member, why do it here? This code is supposed to follow the spec algorithms.